web/client-api/src/main/java/io/deephaven/web/client/api/input/JsInputTable.java

```diff
@@ -60,93 +62,136 @@ public JsInputTable(JsTable from, String[] keys, String[] values) {
+                        if (merged != tablesToAdd[0]) {
+                            // this is an intermediate table for the merge, close it
+                            merged.close();
+                        }
+                        return Promise.reject(failure);
+                    });
+                })
+                .then(response -> Promise.resolve(this));
     }
 
     public Promise<JsInputTable> deleteTable(JsTable tableToDelete) {
         return deleteTables(new JsTable[] {tableToDelete});
     }
 
     public Promise<JsInputTable> deleteTables(JsTable[] tablesToDelete) {
-        final List<String> keysList = Arrays.asList(keys);
-        final List<JsRunnable> cleanups = new ArrayList<>();
-        final BatchBuilder builder = new BatchBuilder();
-
-        // Limit the tables to delete to just the key columns
-        for (int i = 0; i < tablesToDelete.length; i++) {
-            final JsTable tableToDelete = tablesToDelete[i];
-            final BatchBuilder.BatchOp op = builder.getOp();
-            final WorkerConnection connection = tableToDelete.getConnection();
-            final ClientTableState cts = connection.newState(tableToDelete.state(), op);
-            cleanups.add(cts.retain(this));
-
-            // final HandleMapping mapping = new HandleMapping(tableToDelete.getHandle(), cts.getHandle());
-            // op.fromState(cts);
-            // op.setAppendTo(cts.getPrevious());
-            // op.setHandles(mapping);
-            // builder.setViewColumns(keysList);
-            // builder.nextOp(connection, op);
-            throw new UnsupportedOperationException("Can't build batch");
+        if (tablesToDelete.length == 0) {
+            return Promise.resolve(this);
         }
 
-        // TODO core#273
-        BatchTableRequest req = new BatchTableRequest();
-        // req.setOps(builder.serializable());
-
-        return Callbacks.<BatchTableResponse, String>promise(this.table, c -> {
-            // table.getServer().batch(req, c);
-            throw new UnsupportedOperationException("batch");
-        }).then(response -> {
-            if (response.getFailureMessages().length > 0) {
-                return (Promise) Promise
-                        .reject("Unable to delete tables: " + Arrays.toString(response.getFailureMessages()));
+        // for each table, make a view on that table of only key columns, then union the tables and drop together
+        final List<JsRunnable> cleanups = new ArrayList<>();
+        final Ticket ticketToDelete;
+        if (tablesToDelete.length == 1) {
+            JsTable onlyTable = tablesToDelete[0];
+            // don't try too hard to find matching columns, if it looks like we have a match go for it
+            if (onlyTable.getColumns().length == keys.length && onlyTable.findColumns(keys).length == keys.length) {
+                ticketToDelete = onlyTable.getHandle().makeTicket();
+            } else {
+                // view the only table
```

Comment:
Shouldn't it throw if the tablesToDelete doesn't match the schema of the input table you're deleting from?

Reply:
It will throw, but not until later. I can add a local check here to give a clearer message.